### DISCUSSION TITLE
[Discussion] Use a separate emulation backend for loading 32-bit libraries?

### DISCUSSION THREAD

**Discussion by toastmod** [ptitSeb/box64#341] [General]
Hey I had this idea of incorporating qemu-user or unicorn emulator into box64 specifically for loading 32-bit libs. Could be useful for loading Steam without having to use a 32-bit operating system with box86. Might be complicated, but if someone can tell me where the library loading part of box64's source code is I'd like to start looking into it myself.

**Comment by ptitSeb** [comment]
That will make things slow. Also, that defeat the whole twist of box with native lib wrapping.

But if you really want to look: native libraries are loaded at `library.c` mainly. But you also find some loading on `wrappedlibdl.c`. All the "wrapped*.c` file also handle native library wrapping.

I don't think there an easy way to do that. It looks like a huge work.

**Comment by ptitSeb** [comment]
Converted to a discussion, as it is one!
